R/connect.R

```diff
@@ -110,6 +110,12 @@ Connect <- R6::R6Class(
       paste(self$server, "__api__", ..., sep = "/")
     },
 
+    #' @description Build a URL relative to the server root
+    #' @param ... path segments
+    server_url = function(...) {
+      paste(self$server, ..., sep = "/")
+    },
```

Comment:
Any reason to use `server_url` in the call to `api_url`? It might be over-engineering if we really only use it for the (old) thumbnails. But in that case, it might be good to mention that here that this is not really something someone should use, and is only here for compatibility (and when that compatibility is removed so too should this)

Reply:
Oh sorry, actually I have this backwards! The _new_ API has thumbnails at an `__api__`less URL, like `https://connect.company.com/content/25438b83-ea6d-4839-ae8e-53c52ac5f9ce/__thumbnail__` So this helper is needed for those. Are there any other circumstances we would need this? Maybe we should "just" construct that special URL in the thumbnail function rather than exposing here? 

It's definitely confusing to me to have two methods that provide almost the same thing.

Reply:
I don't feel strongly either way. I remember feeling at the time like it was better to create a function that lived alongside `api_url` — I think I discussed it with @nealrichardson at the time. I don't remember the specific reason why I went in this direction. T I can definitely imagine other reasons why we might want this, but they're all hypothetical at this point so kinda moot.

Reply:
If that's the case I would move it down and construct the URLs in `*_thumbnail()` with a comment that we can't use the API url constructors cause those URLs are outside that scope, if we run into that more we might consider factoring it out.

But leaving it here with a heavy comment that we question it's needed is something we could do too. Though I suspect it will still confuse us / we won't remember to look here and see that we (probably) should (almost?) never use it

Reply:
How about:

* Rewrite `api_url()` to be `function(...) self$server_url("__api__", ...)` to make the relationship more clear
* Make sure the use of server_url in the thumbnail function is well commented to call out that it's GETting something that's not under `__api__`

Reply:
@nealrichardson I had considered that; I like it. Done.